src/coreComponents/physicsSolvers/multiphysics/HydrofractureSolver.cpp

```diff
@@ -272,41 +273,32 @@ void HydrofractureSolver::updateDeformationForCoupling( DomainPartition & domain
     separationCoeff0 = subRegion.getReference< array1d< real64 > >( viewKeyStruct::separationCoeff0String() );
 #endif
 
-    forAll< serialPolicy >( subRegion.size(), [=, &contactRelation] ( localIndex const kfe )
+    constitutiveUpdatePassThru( contact, [&] ( auto & castedContact )
     {
-      localIndex const kf0 = elemsToFaces[kfe][0];
-      localIndex const kf1 = elemsToFaces[kfe][1];
-      localIndex const numNodesPerFace = faceToNodeMap.sizeOfArray( kf0 );
-      real64 temp[ 3 ] = { 0 };
-      for( localIndex a=0; a<numNodesPerFace; ++a )
-      {
-        LvArray::tensorOps::add< 3 >( temp, u[ faceToNodeMap( kf0, a ) ] );
-        LvArray::tensorOps::subtract< 3 >( temp, u[ faceToNodeMap( kf1, a ) ] );
-      }
-
-      // TODO this needs a proper contact based strategy for aperture
-      aperture[kfe] = -LvArray::tensorOps::AiBi< 3 >( temp, faceNormal[ kf0 ] ) / numNodesPerFace;
-
-      effectiveAperture[kfe] = contactRelation.effectiveAperture( aperture[kfe] );
-
-
+      using ContactType = TYPEOFREF( castedContact );
+      typename ContactType::KernelWrapper contactWrapper = castedContact.createKernelWrapper();
+
+      HydrofractureSolverKernels::DeformationUpdateKernel
+        ::launch< parallelDevicePolicy<> >( subRegion.size(),
+                                            contactWrapper,
+                                            u,
+                                            faceNormal,
+                                            faceToNodeMap,
+                                            elemsToFaces,
+                                            area,
+                                            volume,
+                                            deltaVolume,
+                                            aperture,
+                                            effectiveAperture
```

Comment:
```suggestion
                                            effectiveAperture,
```

Reply:
I left this as is, to make sure that the code compiles if the `define` has been set